lib/p5.play.js

```diff
@@ -4338,4 +3909,1314 @@ p5.prototype._warn = function(message) {
+    box.updateFromSprite(sprite);
+    return box;
+  };
+
+  /**
+   * Update this collider based on the properties of a parent Sprite.
+   * @method updateFromSprite
+   * @param {Sprite} sprite
+   * @see p5.CollisionShape.prototype.getsDimensionsFromSprite
+   */
+  p5.AxisAlignedBoundingBox.prototype.updateFromSprite = function(sprite) {
+    if (this.getsDimensionsFromSprite) {
+      this._width = sprite.width;
+      this._height = sprite.height;
+    }
+    this.setParentTransform(sprite);
+  };
+
+  /**
+   * Recalculate cached properties, relevant vectors, etc. when at least one
+   * of the shape's transforms changes.  The base CollisionShape (and CollisionPoint)
+   * only need to recompute the shape's center, but other shapes may need to
+   * override this method and do additional recomputation.
+   * @method _onTransformChanged
+   * @protected
+   */
+  p5.AxisAlignedBoundingBox.prototype._onTransformChanged = function() {
+    p5.CollisionShape.prototype._onTransformChanged.call(this);
+    this._computeHalfDiagonals();
+  };
+
+  /**
+   * Recompute this bounding box's half-diagonal vectors.
+   * @method _computeHalfDiagonals
+   * @private
+   * @return {Array.<p5.Vector>}
+   */
+  p5.AxisAlignedBoundingBox.prototype._computeHalfDiagonals = function() {
+    // We transform the rectangle (which may scale and rotate it) then compute
+    // an axis-aligned bounding box _around_ it.
+    var composedTransform = p5.Transform2D.mult(this._parentTransform, this._localTransform);
+    var transformedDiagonals = [
+      new p5.Vector(this._width / 2, -this._height / 2),
+      new p5.Vector(this._width / 2, this._height / 2),
+      new p5.Vector(-this._width / 2, this._height / 2)
+    ].map(function(vertex) {
+      return vertex.transform(composedTransform).sub(this.center);
+    }.bind(this));
+
+    var halfWidth = Math.max(
+      Math.abs(transformedDiagonals[0].x),
+      Math.abs(transformedDiagonals[1].x)
+    );
+    var halfHeight = Math.max(
+      Math.abs(transformedDiagonals[1].y),
+      Math.abs(transformedDiagonals[2].y)
+    );
+
+    this._halfDiagonals = [
+      new p5.Vector(halfWidth, -halfHeight),
+      new p5.Vector(halfWidth, halfHeight)
+    ];
+  };
+
+  /**
+   * Debug-draw this collider.
+   * @method draw
+   * @param {p5} sketch - p5 instance to use for drawing
+   */
+  p5.AxisAlignedBoundingBox.prototype.draw = function(sketch) {
+    sketch.push();
+    sketch.rectMode(sketch.CENTER);
+    sketch.translate(this.center.x, this.center.y);
+    sketch.noFill();
+    sketch.stroke(0, 255, 0);
+    sketch.strokeWeight(1);
+    sketch.rect(0, 0, Math.abs(this._halfDiagonals[0].x) * 2, Math.abs(this._halfDiagonals[0].y) * 2);
+    sketch.pop();
+  };
+
+  /**
+   * Compute candidate separating axes relative to another object.
+   * @method _getCandidateAxes
+   * @protected
+   * @return {Array.<p5.Vector>}
+   */
+  p5.AxisAlignedBoundingBox.prototype._getCandidateAxes = function() {
```

Comment:
Yes, as [documented by the square brackets around the name](http://yui.github.io/yuidoc/syntax/index.html#param).  What's not particularly well-documented is that width and height must be provided together, so the actual signature here is
```
createFromSprite(sprite[, offset[, width, height[, rotation]]])
```

There's some weirdness and poor documentation around what parameters to the constructor (and parent constructor) are optional though - perhaps I should do some cleanup around that.